classes/Dispatcher.php

```diff
@@ -315,6 +306,38 @@ private function setDefaultController($defaultController)
         $this->default_controller = $defaultController;
     }
 
+    /**
+     * Sets the front controller type.
+     */
+    private function setFrontControllerType()
+    {
+        $this->front_controller = self::FC_FRONT;
+
+        if (defined('_PS_ADMIN_DIR_')) {
+            $this->front_controller = self::FC_ADMIN;
+        }
+        
+        if ('module' === Tools::getValue('fc')) {
+            $this->front_controller = self::FC_MODULE;
+        }
```

Comment:
```
if (defined('_PS_ADMIN_DIR_')) {
    $this->front_controller = self::FC_ADMIN;
} elseif ('module' === Tools::getValue('fc')) {
    $this->front_controller = self::FC_MODULE;
} else {
    $this->front_controller = self::FC_FRONT;
}
```
Is maybe easier to read and prevent useless tests (no need to test elseif or else if if is true :D